### PR TITLE
fix: switch from progress to message for init

### PR DESCRIPTION
**Pull request by sergiusens**
- Snapcraft uses message, progress hides the docs url as it is "uni-line"
- Use a similar message to what is done in Snapcraft
- Improve the test for initializing flask-framework
- Use the correct documentation slug

- [ ] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----

### CODE REVIEW

**Review by cmatsuoka — APPROVED** [review]

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
ty.

maybe you could use the new doc_slug variable here as well: https://github.com/canonical/rockcraft/pull/604/files#diff-a065a85026852eac577f63c3df7d7502eb85371b5ced140674a9e9b66d6bdaadR110 so we don't have the boilerplate text.

Another question, probably a bit more controversial, is about `stable` vs `latest`. Whoever is using `latest/edge` will be getting doc links that might not exist :/

Should the doc_slugs be based on the rockcraft version?

**Comment by tigarmo** [comment]
I'm fine with the idea in principle (though we'd probably do it in craft-application instead). The problem is that by default new versions on readthedocs are private - that is, if I'm logged out I can access https://documentation.ubuntu.com/rockcraft/en/stable/ but not https://documentation.ubuntu.com/rockcraft/en/1.4.0/

And I can't seem to find a setting to change this default visibility.

File: tests/unit/test_cli.py [hunk 1]
```diff
@@ -131,7 +131,12 @@ def test_run_init_flask(mocker, tmp_path, monkeypatch):
 
     assert len(rock_project_yaml["summary"]) < 80
     assert len(rock_project_yaml["description"].split()) < 100
-    assert mock_ended_ok.mock_calls == [call()]
-
+    emitter.assert_message(
+        textwrap.dedent(
+            """\
+        Created 'rockcraft.yaml'.
```
Comment (tigarmo) [inline tests/unit/test_cli.py:137]: do you specifically want the single quotes around rockcraft.yaml?

**Comment by sergiusens** [comment]
@cjdcordeiro yes to what you say, but not now. We discussed this and we decided to do it after the rush. Also, not part of this PR per se.

File: tests/unit/test_cli.py (hunk 1 above)
Comment (sergiusens) [reply tests/unit/test_cli.py:137]: I did say I am following `snapcraft init`, which produces:

    Created 'snap/snapcraft.yaml'.
    Go to https://docs.snapcraft.io/the-snapcraft-format/8337 for more information about the snapcraft.yaml format.

**Comment by sergiusens** [comment]
@cjdcordeiro fwiw, I do want versioned slugs :-)

**Comment by cjdcordeiro** [comment]
@sergiusens fair enough. Let's then at least parameterize the doc_url in the YAML text. 

Do you already have an issue somewhere with the "versioned slugs" task? (ack @tigarmo , I didn't know that either. can you please raise that issue in the `#documentation` channel? Btw, I can manually edit the 1.4.0 visibility and make it public, but indeed, I see that some versions are public and some are private...and I don't know if there's a way to automatically set them all to public)

**Comment by sergiusens** [comment]
@cjdcordeiro just did :-)

**Comment by tigarmo** [comment]
> can you please raise that issue in the #documentation channel?

Sure, I'll do that on monday. Since we have readthedocs for business we can probably raise a support ticket

**Review by tigarmo — APPROVED** [review]

**Pull request merged**